File: problem-solving-swift/leetcode/twoSum.swift
//
//  twoSum.swift
//  problem-solving-swift
//
//  Created by Avijeet on 05/04/24.
//

import Foundation

class TwoSum {
    static func findIndices(_ nums: [Int], _ target: Int) -> [Int] {
        
        for i in 0..<nums.count {
            for j in i+1..<nums.count {
                if nums[i] + nums[j] == target {
                    return [i,j]
                }
            }
        }
        
        return []
    }
    
    static func findIndicesMap(_ nums: [Int], _ target: Int) -> [Int] {
        var frequencyMap: [Int: Int] = [:]
        
        for i in 0..<nums.count {
            let difference = target - nums[i]
            
            if let differenceIndex = frequencyMap[difference] {
                return [i, differenceIndex]
            }
            
            frequencyMap[nums[i]] = i
        }
        
        return []
    }
}

File: problem-solving-swift/leetcode/twoSum.js
function twoSum(arr, target) {
    for(let i=0;i<arr.length;i++) {
        for(let j=i+1;j<arr.length;j++) {
            if(arr[i]+arr[j] === target) {
                return [i,j]
            }
        }
    }
    
    return []
}

let arr = [2,4,5,6,9]

console.log(twoSum(arr,6))
